### ISSUE TITLE
Got error when train SDXL-Dreambooth

### ISSUE THREAD

**Issue by helpmeIamnewbie** [huggingface/diffusers#4256]
### Describe the bug

after text_encoder fine-tune branch is merged, I got error in run train_dreambooth_lora_sdxl.py.

### Reproduction

python = "3.10.12"
wandb = "0.15.5"
xformers = "0.0.20"
transformers = "4.28.1"
diffusers = { git = "https://github.com/huggingface/diffusers.git" }
accelerate = "0.21.0"
streamlit = "1.24.1"
openai = "0.27.8"
fastapi = "0.100.0"
uvicorn = "0.22.0"
google-cloud-storage = "2.10.0"
google-cloud-bigquery = "3.11.3"
google-auth = "2.22.0"
google-cloud-error-reporting = "1.9.2"
torch = { url = "https://download.pytorch.org/whl/cu118/torch-2.0.1%2Bcu118-cp310-cp310-linux_x86_64.whl" }
torchvision = { url = "https://download.pytorch.org/whl/cu118/torchvision-0.15.2%2Bcu118-cp310-cp310-linux_x86_64.whl" }
httpx = "0.24.1"
rpds-py = "0.8.10"
pyyaml = "6.0"
apache-airflow = "2.6.3"
apache-airflow-providers-google = "10.3.0"
apache-airflow-providers-amazon = "8.3.0"
apache-airflow-providers-apache-beam = "5.1.1"
apache-airflow-providers-apache-cassandra = "3.2.1"
apache-airflow-providers-cncf-kubernetes = "7.2.0"
apache-airflow-providers-common-sql = "1.6.0"
apache-airflow-providers-facebook = "3.2.1"
apache-airflow-providers-microsoft-azure = "6.2.0"
apache-airflow-providers-microsoft-mssql = "3.4.1"
apache-airflow-providers-mysql = "5.1.1"
apache-airflow-providers-oracle = "3.7.1"
apache-airflow-providers-postgres = "5.5.2"
apache-airflow-providers-presto = "5.1.1"
apache-airflow-providers-salesforce = "5.4.1"
redis = "4.6.0"
celery = "5.3.1"
ftfy = "6.1.1"
Jinja2 = "3.1.2"
tensorboard = "2.13.0"
invisible-watermark = "0.2.0"

### Logs

```shell
Traceback (most recent call last):
  File "/opt/ml/level3_cv_finalproject-cv-03/src/stable_diffusion_xl/train.py", line 1351, in <module>
    main(args)
  File "/opt/ml/level3_cv_finalproject-cv-03/src/stable_diffusion_xl/train.py", line 1102, in main
    model_pred = unet(
  File "/opt/ml/.cache/pypoetry/virtualenvs/final-project-MxwomGDb-py3.10/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/opt/ml/.cache/pypoetry/virtualenvs/final-project-MxwomGDb-py3.10/lib/python3.10/site-packages/accelerate/utils/operations.py", line 581, in forward
    return model_forward(*args, **kwargs)
  File "/opt/ml/.cache/pypoetry/virtualenvs/final-project-MxwomGDb-py3.10/lib/python3.10/site-packages/accelerate/utils/operations.py", line 569, in __call__
    return convert_to_fp32(self.model_forward(*args, **kwargs))
  File "/opt/ml/.cache/pypoetry/virtualenvs/final-project-MxwomGDb-py3.10/lib/python3.10/site-packages/torch/amp/autocast_mode.py", line 14, in decorate_autocast
    return func(*args, **kwargs)
  File "/opt/ml/.cache/pypoetry/virtualenvs/final-project-MxwomGDb-py3.10/lib/python3.10/site-packages/diffusers/models/unet_2d_condition.py", line 874, in forward
    emb = emb + aug_emb if aug_emb is not None else emb
RuntimeError: The size of tensor a (8) must match the size of tensor b (16) at non-singleton dimension 0
Steps:   0%|                                                                                                                                                                            | 0/1 [00:02<?, ?it/s]
Traceback (most recent call last):
  File "/opt/ml/.cache/pypoetry/virtualenvs/final-project-MxwomGDb-py3.10/bin/accelerate", line 8, in <module>
    sys.exit(main())
  File "/opt/ml/.cache/pypoetry/virtualenvs/final-project-MxwomGDb-py3.10/lib/python3.10/site-packages/accelerate/commands/accelerate_cli.py", line 45, in main
    args.func(args)
  File "/opt/ml/.cache/pypoetry/virtualenvs/final-project-MxwomGDb-py3.10/lib/python3.10/site-packages/accelerate/commands/launch.py", line 979, in launch_command
    simple_launcher(args)
  File "/opt/ml/.cache/pypoetry/virtualenvs/final-project-MxwomGDb-py3.10/lib/python3.10/site-packages/accelerate/commands/launch.py", line 628, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['/opt/ml/.cache/pypoetry/virtualenvs/final-project-MxwomGDb-py3.10/bin/python', 'train.py', '--pretrained_model_name_or_path=stabilityai/stable-diffusion-xl-base-0.9
```

### System Info

ubuntu

**Comment by patrickvonplaten** [comment]
cc @sayakpaul

**Comment by sayakpaul** [comment]
I am able to run from `main`. See the run page with all the details: https://wandb.ai/sayakpaul/dreambooth-lora-sd-xl/runs/rq1h4o1d/overview.

I used the following command:

```python
accelerate launch train_dreambooth_lora_sdxl.py \
  --pretrained_model_name_or_path=$MODEL_NAME  \
  --pretrained_vae_model_name_or_path="madebyollin/sdxl-vae-fp16-fix" \
  --instance_data_dir=$INSTANCE_DIR \
  --output_dir=$OUTPUT_DIR \
  --mixed_precision="fp16" \
  --instance_prompt="a photo of sks dog" \
  --resolution=1024 \
  --train_batch_size=1 \
  --gradient_accumulation_steps=4 \
  --learning_rate=1e-4 \
  --report_to="wandb" \
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --max_train_steps=100 \
  --train_text_encoder \
  --validation_prompt="A photo of sks dog in a bucket" \
  --validation_epochs=25 \
  --seed="0" \
  --push_to_hub
```

Could you tell me what version of `diffusers` you're using and what training command?

**Comment by helpmeIamnewbie** [comment]
my version in poetry is "version = "0.19.0.dev0" and I'm using command
accelerate launch train.py 
--pretrained_model_name_or_path=stabilityai/stable-diffusion-xl-base-0.9 
--pretrained_vae_model_name_or_path=madebyollin/sdxl-vae-fp16-fix 
--instance_data_dir=/opt/ml/stable-diffusion-xl/data/tokens/qwer 
--class_data_dir=/opt/ml/stable-diffusion-xl/data/generated/woman 
--output_dir=/opt/ml/stable-diffusion-xl/weights/qwer/num200 
--with_prior_preservation 
--prior_loss_weight=1.0 
--mixed_precision=fp16 
--instance_prompt="A photo of qwer woman" 
--class_prompt="A photo of a woman" 
--resolution=512 
--train_batch_size=4 
--gradient_accumulation_steps=1 
--gradient_checkpointing 
--learning_rate=1e-4 
--lr_scheduler=constant 
--lr_warmup_steps=0 
--max_train_steps=1 
--num_class_images=200 
--sample_batch_size=1 
--seed=1557 
--train_text_encoder

**Comment by sayakpaul** [comment]
Could you maybe do the following?

* Run `git pull`. 
* Uninstall diffusers and reinstall it with `pip install -e .`? 

For me to reproduce your issue, I need to have access to your dataset. Could you maybe try to reproduce it with the `dogs` dataset that we mention in the `README_sdxl.md` file?

**Comment by a-sane** [comment]
I have same error with 0.19.0.dev0 diffusers, looks like `--with_prior_preservation` option leads to this error

**Comment by sayakpaul** [comment]
Let me try to debug :)

**Comment by helpmeIamnewbie** [comment]
I'm using poetry and I use 
diffusers = { git = "https://github.com/huggingface/diffusers.git" }
in my pyproject.toml files. thanks for your answer and I also think error occurs in `--with_prior_preservation` option.

**Comment by sayakpaul** [comment]
Currently working on a fix here: #4262.
